bot/cogs/filtering.py

```diff
@@ -360,6 +401,34 @@ def mod_log(self) -> ModLog:
             await channel.send(f"{filtered_member.mention} {reason}")
 
 
+async def delete_offensive_msg(bot: Bot) -> None:
+    """Background task that pull up a list of offensive messages every day and delete them."""
+    while True:
+        tomorrow = datetime.date.today() + datetime.timedelta(days=1)
+        time_until_next = datetime.datetime(tomorrow.year, tomorrow.month, tomorrow.day) - datetime.datetime.now()
+        try:
+            msg_list = await bot.api_client.get(
+                'bot/offensive-message',
+                params={'delete_date': datetime.date.today().isoformat()}
+            )
+        except ResponseCodeError as e:
+            log.error(f"Failed to get offending messages to delete (got code {e.response.status}), "
+                      f"retrying in 30 minutes.")
+            time_until_next = datetime.timedelta(minutes=30)
+            msg_list = []
+        for msg in msg_list:
+            try:
+                channel = bot.get_channel(msg['channel_id'])
+                if channel:
+                    msg_obj = await channel.fetch_message(msg['id'])
+                    await msg_obj.delete()
+            except NotFound:
+                log.info(f"Tried to delete message {msg['id']}, but the message can't be found "
+                         f"(it has been probably already deleted).")
+        log.info(f"Deleted {len(msg_list)} offensive message(s).")
+        await asyncio.sleep(time_until_next.seconds)
```

Comment:
I know the original issue spelled out `daily`, but we already have a base class (`Scheduler`) that allows us to schedule "something" at a time point in the future. We already use it for infractions and reminders, so I'd rather have this feature use that as well, to avoid having all kinds of different "scheduler"-like logic around the bot. Using a single tried and tested method makes the most sense to me.